rest-service/manager_rest/rest/resources_v3_1/blueprints.py

```diff
@@ -200,3 +230,17 @@ def put(self, blueprint_id, **kwargs):
         return UploadedBlueprintsValidator().\
             receive_uploaded_data(data_id=blueprint_id,
                                   visibility=visibility)
+
+
+class BlueprintsIdArchive(resources_v1.BlueprintsIdArchive):
+    @authorize('blueprint_upload')
+    def put(self, blueprint_id, **kwargs):
+        """
+        Upload an archive for an existing a blueprint.
+
+        Used for uploading the blueprint's archive, downloaded from a URL using
+        a system workflow, to the manager's file server
+        This method is for internal use only.
+        """
+        UploadedBlueprintsManager(). \
```

Comment:
@YoniItzhak yes, all fileserver uploads etc must go through the restservice, and shared filesystem access between the mgmtworker and other parts (eg. putting a file into the fileserver dirs, from inside the mgmtworker) is prohibited, because that would
1) require they even have a common filesystem (not necessarily the case with a future one-service-per-container split)
2) make the mgmtworker more stateful than necessary